core-bundle/src/EventListener/SearchIndexListener.php

```diff
@@ -97,6 +97,18 @@ private function needsIndex(Request $request, Response $response, Document $docu
             return false;
         }
 
+        // searchIndexer setting has priority over robots tag
+        $pageJsonLds = $document->extractJsonLdScripts('https://schema.contao.org/', 'Page');
+        $pageSearchIndexer = $pageJsonLds[0]['searchIndexer'] ?? null;
+
+        if ('always_index' === $pageSearchIndexer) {
+            return true;
+        }
+
+        if ('never_index' === $pageSearchIndexer) {
+            return false;
+        }
```

Comment:
This should be moved above the `X-Robots-Tag` check as responses with `X-Robots-Tag: noindex` and `"searchIndexer": "always_index",` should still be indexed I think.

Reply:
See also https://github.com/contao/contao/pull/8252#issuecomment-2761678313

Reply:
> See also [#8252 (comment)](https://github.com/contao/contao/pull/8252#issuecomment-2761678313)

I don’t agree with that, but feel free to overrule and resolve this thread ☺️

Reply:
> Technically X-Robots-Tag and <meta name=robots> are interchangeable so I think we should handle them the same way.

Yes, but the `X-Robots-Tag` is not user-editable. If a controller decides to set `X-Robots-Tag`, I think it should take precedence and not be overridable by the user.

> We actually don’t know where a X-Robots-Tag can come from I think as Contao does not set this header AFAIK.

We do - for all back end requests, as they should never appear in any search index (see `BackendNoindexListener`). And I think it would make sense for a `PageController` to be able to do the same.

We also do it in our `preview.php` entry point. If the `X-Robots-Tag` does not take precedence over `searchIndex`, then `preview.php` URLs would suddenly be indexable.